File: CodeVita2020/B.py
try:
    n=int(input())
    # input()
    lat=list(map(float,input().split()))
    # input()
    long=list(map(float,input().split()))
    # input()
    height=list(map(int,input().split()))
    # input()
    jason=list(map(float,input().split()))
    # input()
    radius=6371
    tower=list(zip(lat,long,height))

    def reach(heig):
        val=2*heig*radius
        dist=pow(val,0.5)
        return dist


    from math import radians, cos, sin, asin, sqrt


    def distance(lat1, lat2, lon1, lon2):
        lon1 = radians(lon1)
        lon2 = radians(lon2)
        lat1 = radians(lat1)
        lat2 = radians(lat2)

        # Haversine formula
        dlon = lon2 - lon1
        dlat = lat2 - lat1
        a = sin(dlat / 2) ** 2 + cos(lat1) * cos(lat2) * sin(dlon / 2) ** 2
        c = 2 * asin(sqrt(a))
        return c * radius

    def possible(tow,jas):
        x=reach(tow[2])
        y=distance(tow[0],jas[0],tow[1],jas[1])
        return y<=x

    count=0
    for i in range(n):
        if possible(tower[i],jason):
            count+=1
    print(count)
except :
    pass

File: CodeVita2020/B.java
import java.util.*;

class B
{
    Scanner in = new Scanner(System.in);
    int n = in.nextInt(), radius = 6371;
    double lat[] = new double[n];
    double lng[] = new double[n];
    int height[] = new int[n];
    double jlat, jlong;
    void solve(){
        for(int i=0; i<n; i++)
        lat[i] = in.nextDouble();
        for(int i=0; i<n; i++)
        lng[i] = in.nextDouble();
        for(int i=0; i<n; i++)
        height[i] = in.nextInt();
        jlat = in.nextDouble();
        jlong = in.nextDouble();
        int ans= 0;
        for(int i=0; i<n; i++){
            double val1 = dist(lat[i], lng[i]);
            double val2 = Math.pow(2*height[i]*radius,0.5);
            if(Double.compare(val1,val2)<=0)
            ans++;
        }
        System.out.print(ans);
    }
    double dist(double x1, double y1) 
    {
        double d1 = Math.toRadians(x1 - jlat); 
        double d2 = Math.toRadians(y1 - jlong);
        x1 = Math.toRadians(x1); 
        double x3 = Math.toRadians(jlat); 
        double ab = Math.pow(Math.sin(d1 / 2), 2) + Math.pow(Math.sin(d2 / 2), 2) *  
                   Math.cos(x1) *  Math.cos(x3);
        return radius *2 * Math.asin(Math.sqrt(ab));
    } 
    public static void main(String args[]){
        new B().solve();
    }
}